src/Uno.UI.Composition/Composition/Uno/VisualExtensions.cs

```diff
@@ -0,0 +1,16 @@
+using System.Numerics;
+
+namespace Microsoft.UI.Composition;
+
+internal static class VisualExtensions
+{
+	internal static Vector3 GetTotalOffset(this Visual visual)
+	{
+		if (visual.IsTranslationEnabled && visual.Properties.TryGetVector3("Translation", out var translation) == CompositionGetValueStatus.Succeeded)
```

Comment:
Well in order to keep the `Properties` lazily init, we should avoid to use the `Properties` property here. A solution could be to add an internal `TryGetProperty` on the `Visual` (which almost breaks the benefit to have this as an extensions method ... your choice ^^)

Reply:
@dr1rrb Properties is only accessed if IsTranslationEnabled is true. The default is false.